spec.html

```diff
@@ -7018,6 +7052,223 @@ <h1>
+            <td>
+              ~sync-dispose~
+            </td>
+            <td>
+              Indicates that the resource was added by a `using` declaration or DisposableStack object.
+            </td>
+          </tr>
+          <tr>
+            <td>
+              [[DisposeMethod]]
+            </td>
+            <td>
+              a function object
+            </td>
+            <td>
+              A function object that will be called with [[ResourceValue]] as its *this* value when the resource disposed.
+            </td>
+          </tr>
+        </table>
+      </emu-table>
+      <emu-note>
+        Currently, the only allowed value for [[Hint]] is ~sync-dispose~.
+      </emu-note>
+    </emu-clause>
+
+    <emu-clause id="sec-newdisposecapability" type="abstract operation">
+      <h1>NewDisposeCapability ( ): a DisposeCapability Record</h1>
+      <dl class="header">
+      </dl>
+      <emu-alg>
+        1. Let _stack_ be a new empty List.
+        1. Return the DisposeCapability Record { [[DisposableResourceStack]]: _stack_ }.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-adddisposableresource" type="abstract operation">
+      <h1>
+        AddDisposableResource (
+          _disposeCapability_: a DisposeCapability Record,
+          _V_: an ECMAScript language value,
+          _hint_: ~sync-dispose~,
+          optional _method_: an ECMAScript language value,
+        ): either a normal completion containing ~unused~ or a throw completion
+      </h1>
+      <dl class="header">
+      </dl>
+      <emu-alg>
+        1. If _method_ is not present, then
+          1. If _V_ is either *null* or *undefined*, return ~unused~.
+          1. If _V_ is not an Object, throw a *TypeError* exception.
+          1. Let _resource_ be ? CreateDisposableResource(_V_, _hint_).
+        1. Else,
+          1. If _V_ is either *null* or *undefined*, then
+            1. Let _resource_ be ? CreateDisposableResource(*undefined*, _hint_, _method_).
+          1. Else,
+            1. If _V_ is not an Object, throw a *TypeError* exception.
+            1. Let _resource_ be ? CreateDisposableResource(_V_, _hint_, _method_).
+        1. Append _resource_ to _disposeCapability_.[[DisposableResourceStack]].
+        1. Return ~unused~.
+      </emu-alg>
+      <emu-note>
+        Currently, the only allowed value for _hint_ is ~sync-dispose~.
+      </emu-note>
+    </emu-clause>
+
+    <emu-clause id="sec-createdisposableresource" type="abstract operation">
+      <h1>
+        CreateDisposableResource (
+          _V_: an Object or *undefined*,
+          _hint_: ~sync-dispose~,
+          optional _method_: an ECMAScript language value,
+        ): either a normal completion containing a DisposableResource Record or a throw completion
+      </h1>
+      <dl class="header">
+      </dl>
+      <emu-alg>
+        1. If _method_ is not present, then
+          1. If _V_ is *undefined*, throw a *TypeError* exception.
```

Comment:
This never happens, because the only call to `CreateDisposableResource` without the third parameter already validates _V_.

Reply:
`using` does not need this, but `await using` does. Now that they have merged, this needs to remain.